core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt

```diff
@@ -96,11 +97,16 @@ internal class SyncStream(
 //                    break;
 //                }
             } catch (e: Exception) {
-                // If the coroutine was cancelled, don't log an error
-                if (e !is CancellationException) {
-                    logger.e(e) { "Error in streamingSync" }
+                if (e is CancellationException) {
+                    throw e
                 }
-                invalidCredentials = true
+
+                logger.e(e) { "Error in streamingSync: $e" }
+                // If there is a network issue don't invalidate the credentials
+                if (e !is IOException) {
```

Comment:
Could we match the Dart implementation and invalidate if we [receive](https://github.com/powersync-ja/powersync.dart/blob/bd22580265eb1b690d19e629c421f3233c17adf1/packages/powersync_core/lib/src/streaming_sync.dart#L263) a 401 HTTP response code or if we receive a message that the token is expiring [soon](https://github.com/powersync-ja/powersync.dart/blob/bd22580265eb1b690d19e629c421f3233c17adf1/packages/powersync_core/lib/src/streaming_sync.dart#L408)?

Reply:
The 401 invalidation is happening here https://github.com/powersync-ja/powersync-kotlin/blob/c74857ea9a155c71d070e978d3a2b4173daca6d6/core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt#L178 and the token expiring invalidation is happening here https://github.com/powersync-ja/powersync-kotlin/blob/c74857ea9a155c71d070e978d3a2b4173daca6d6/core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt#L393

Reply:
If we already invalidate in those places, what is the reason for invalidating in this section of the code?

Reply:
Not sure, it seems to have been added in the beginning probably before the other invalidations were added. I will remove it.